### ISSUE TITLE
Rule proposal: `no-canvas-to-image`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1271]
Many people don't know [`CanvasRenderingContext2D.drawImage()`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawImage) accept `HTMLCanvasElement` as `CanvasImageSource`

## Fail

```js
const image = await loadImage(canvas1.toDataURL());
context2.drawImage(image);
```

```js
const imageData = context1.getImageData();
context2.putImageData(imageData);
```

## Pass

```js
context2.drawImage(canvas1)
```
